app/connectors/ApiConnector.scala

```diff
@@ -16,80 +16,73 @@
+    with Logging {
 
   private val requestHeaders = Seq(
     HeaderNames.ACCEPT       -> "application/vnd.hmrc.2.0+json",
     HeaderNames.CONTENT_TYPE -> "application/xml"
   )
 
-  // TODO - build out for remaining sections
-  def createSubmission(userAnswers: UserAnswers): Either[String, String] =
+  def createPayload(userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Future[Either[ReaderError, CC015CType]] =
     for {
-      transitOperation <- Conversions.transitOperation(userAnswers)
-    } yield payloadXml(transitOperation)
-
-  def payloadXml(transitOperation: TransitOperationType06): String =
-    (<ncts:CC015C PhaseID="NCTS5.0" xmlns:ncts="http://ncts.dgtaxud.ec">
-      <messageRecipient>3pekcCFaMGmCMz1CPGUlhyml9gJCV6</messageRecipient>
-      <preparationDateAndTime>2022-07-02T03:11:04</preparationDateAndTime>
-      <messageIdentification>wrxe</messageIdentification>
-      <messageType>CC015C</messageType>
-      {toXML[TransitOperationType06](transitOperation, "TransitOperation", generated.defaultScope)}
-      <CustomsOfficeOfDeparture>
-        <referenceNumber>GB000218</referenceNumber>
-      </CustomsOfficeOfDeparture>
-      <CustomsOfficeOfDestinationDeclared>
-        <referenceNumber>GB000218</referenceNumber>
-      </CustomsOfficeOfDestinationDeclared>
-      <HolderOfTheTransitProcedure>
-        <identificationNumber>ezv3Z</identificationNumber>
-      </HolderOfTheTransitProcedure>
-      <Guarantee>
-        <sequenceNumber>66710</sequenceNumber>
-        <guaranteeType>P</guaranteeType>
-        <otherGuaranteeReference>iNkM2E</otherGuaranteeReference>
-      </Guarantee>
-      <Consignment>
-        <grossMass>4380979244.527545</grossMass>
-        <HouseConsignment>
-          <sequenceNumber>66710</sequenceNumber>
-          <grossMass>4380979244.527545</grossMass>
-          <ConsignmentItem>
-            <goodsItemNumber>34564</goodsItemNumber>
-            <declarationGoodsItemNumber>25</declarationGoodsItemNumber>
-            <Commodity>
-              <descriptionOfGoods>fds9YFrlk6DX7pnwQNgJmksfZ4z9uGjDy6Kaucb13r3kEleTuLHD5zKtbAKUU005AaZeVdTgdAnJKzuGliZGRb1E83Y0Z8IuyeFfnXgT7NwX81eGFb3vRXAWUFswwwprqZBcffnBLwLObF45W7evl7C6J4Tihj1d1a2ZKcAU6ttLNy</descriptionOfGoods>
-            </Commodity>
-            <Packaging>
-              <sequenceNumber>66710</sequenceNumber>
-              <typeOfPackages>Nu</typeOfPackages>
-            </Packaging>
-          </ConsignmentItem>
-        </HouseConsignment>
-      </Consignment>
-    </ncts:CC015C>).mkString
+      countries <- countriesService.getCountryCodesCTC()
+      security  <- countriesService.getCustomsSecurityAgreementAreaCountries()
+      domain                  = UserAnswersReader[DepartureDomain](userAnswersReader(countries.countryCodes, security.countryCodes)).run(userAnswers)
+      reducedDatasetIndicator = userAnswers.get(ApprovedOperatorPage).getOrElse(false) // TODO - can we get this from the domain models?
```

Comment:
You may be able to do `ApprovedOperatorPage.inferredReader.run(userAnswers)` here.

Reply:
@jackheslop96  - I wonder how we might add this to the Consignment Domain without breaking the journey? It is a fields that ca be answered as well as inferred and we don't want to redirect a user here where it has been inferred?